src/app/reporting/ReportSchedulerImpl.cpp

```diff
@@ -77,19 +81,27 @@ void ReportSchedulerImpl::OnBecameReportable(ReadHandler * aReadHandler)
 {
     ReadHandlerNode * node = FindReadHandlerNode(aReadHandler);
     VerifyOrReturn(nullptr != node);
+
+    // Update the now timestamp to the current time to calculate the next report timeout
+    Timestamp now = mTimerDelegate->GetCurrentMonotonicTimestamp();
+
     Milliseconds32 newTimeout;
-    CalculateNextReportTimeout(newTimeout, node);
-    ScheduleReport(newTimeout, node);
+    CalculateNextReportTimeout(newTimeout, node, now);
+    ScheduleReport(newTimeout, node, now);
 }
 
 void ReportSchedulerImpl::OnSubscriptionReportSent(ReadHandler * aReadHandler)
 {
     ReadHandlerNode * node = FindReadHandlerNode(aReadHandler);
     VerifyOrReturn(nullptr != node);
-    node->SetIntervalTimeStamps(aReadHandler);
+
+    // Update the now timestamp to the current time to update the node's interval timestamps and calculate the next report timeout
```

Comment:
Though I do wonder how useful these "use the current time" comments are; it's very clear we are using the current time....

Reply:
Right, they were more relevant when there was an mNow timestamp that needed updating at specific times. I`ll just go ahead and remove all of them.